evap/static/ts/src/csrf-utils.ts

```diff
@@ -28,6 +32,27 @@ $.ajaxSetup({
     },
 });
 
+function addCsrfToken(this: HTMLFormElement, ev: Event) {
+    const inputElement = document.createElement("input");
+    inputElement.setAttribute("type", "hidden");
+    inputElement.setAttribute("name", "csrfmiddlewaretoken");
+    inputElement.setAttribute("value", csrftoken);
+
+    const { submitter } = ev as SubmitEvent;
+    const action = (submitter && submitter.getAttribute("formaction")) || this.getAttribute("action");
+    const method = (submitter && submitter.getAttribute("formmethod")) || this.getAttribute("method");
+
+    if (method && !isMethodCsrfSafe(method) && action && !isUrlCrossDomain(action)) {
+        this.insertAdjacentElement("afterbegin", inputElement.cloneNode() as Element);
```

Comment:
Since we now create the `inputElement` on each submit, we don't need to clone anymore; the creation of the element can also be moved into this branch :)

There is one problem though: if we have forms that get submitted multiple times without the site reloading, we insert multiple inputs for the csrf token. One such example view would be the refactor planned in #1698.

I think using the `formdata` event would not have this problem, but I don't think we can check for submitter and target origin in the `formdata` handler sooo :man_shrugging:

Reply:
I added an explicit check.